### PR TITLE
[Bugfix] race condition on fe/be/cn status when upgrade

### PR DESCRIPTION
**Pull request by yandongxiao**

### CODE REVIEW

File: pkg/sub_controller/cn/cn_controller.go [hunk 1]
```diff
@@ -240,7 +241,8 @@ func (cc *CnController) UpdateStatus(src *srapi.StarRocksCluster) error {
 	cs.ServiceName = srapi.GetExternalServiceName(src.Name, cnSpec)
 	cs.ResourceNames = rutils.MergeSlices(cs.ResourceNames, []string{statefulSetName})
 
-	if err := cc.updateCnStatus(cs, pod.Labels(src.Name, src.Spec.StarRocksCnSpec), src.Namespace, *st.Spec.Replicas); err != nil {
+	if err := sub_controller.UpdateStatefulSetStatus(&cs.StarRocksComponentStatus, cc.k8sclient,
```
[the remark this answers is not in the document]
Comment (dengliu) [reply pkg/sub_controller/cn/cn_controller.go:244]: fyi, here is the corresponding cli to check rollout status of deployment
` kubectl rollout status -n ${TEST_NAMESPACE} deployment/starrocks-cn`

[the remark this answers is not in the document]
Comment (yandongxiao) [reply pkg/sub_controller/cn/cn_controller.go:244]: currentlly, CN is statefulset like be.

File: pkg/k8sutils/templates/statefulset/status.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package statefulset
+
+import (
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+)
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply pkg/k8sutils/templates/statefulset/status.go]: done

File: pkg/sub_controller/cn/cn_controller.go [hunk 3]
```diff
@@ -25,6 +25,7 @@ import (
 	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/k8sutils/templates/pod"
 	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/k8sutils/templates/service"
 	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/k8sutils/templates/statefulset"
+	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/sub_controller"
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply pkg/sub_controller/cn/cn_controller.go:30]: done

File: pkg/sub_controller/subcontroller.go [hunk 4]
```diff
@@ -21,6 +21,12 @@ package sub_controller
 import (
 	"context"
 	srapi "github.com/StarRocks/starrocks-kubernetes-operator/pkg/apis/starrocks/v1"
+	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/k8sutils/templates/pod"
+	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/k8sutils/templates/statefulset"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply pkg/sub_controller/subcontroller.go:30]: done

File: pkg/k8sutils/k8sutils_test.go [hunk 5]
```diff
@@ -16,6 +16,8 @@ package k8sutils
 
 import (
 	"context"
+	"testing"
```
Comment (kevincai) [inline pkg/k8sutils/k8sutils_test.go:19]: shall submit a separate PR to format these import lines

**Review by kevincai — APPROVED** [review]
